Stardrop/Views/MainWindow.axaml.cs

```diff
@@ -1861,6 +1887,8 @@ private void DeleteMod(Mod mod)
 
         private async Task<List<Mod>> AddMods(string[]? filePaths)
         {
+            await HandleModListRefresh();
```

Comment:
I believe the problem I ran into was the following:
The list of mods (and their versions) when updating mods may not be current. There may be mods which were added/removed/modified manually. If the refresh of mod list was not run manually since then, when you run `File->Add mods`, the manager will, for example, still ask whether you want to replace old mod files when in fact you have already manually removed the mod files. Also, if you remove the mod from `Stardew Valley/Mods` folder, and you want to reinstall the mod in the Stardrop install folder, you will end up with the mod being reinstalled in the `Stardew Valley/Mods` folder again (since Stardrop thinks you are replacing the original mod instead of installing a "new mod").

I think that in general, when you are planning on updating mods (= writing on disk), you should have the latest information about the files on the disk. Hence, the forced refresh (updating Stardrop's internal list of currently existing mods in the mod folders).